File: app/src/main/java/com/example/tetrisapp/interfaces/PieceConfiguration.java
package com.example.tetrisapp.interfaces;

import com.example.tetrisapp.model.game.Piece;

public interface PieceConfiguration {
    Piece[] getPieces();

    String[] getNames();

    Piece get(String name);

    String[] getStarterPieces();

    String[] getInitialHistory();
}

File: app/src/main/java/com/example/tetrisapp/interfaces/PieceConfiguration.kt
package com.example.tetrisapp.interfaces

import com.example.tetrisapp.model.game.Piece

interface PieceConfiguration {
    val pieces: Array<Piece>
    val names: Array<String>
    val starterPieces: Array<String>
    val initialHistory: Array<String>
    operator fun get(name: String): Piece
}